pkgs/development/interpreters/python/python-packages-base.nix

```diff
@@ -14,27 +14,24 @@ let
 
   # Derivations built with `buildPythonPackage` can already be overridden with `override`, `overrideAttrs`, and `overrideDerivation`.
   # This function introduces `overridePythonAttrs` and it overrides the call to `buildPythonPackage`.
+  #
+  # Overridings specified through `overridePythonAttrs` will always be applied
+  # before those specified by `overrideAttrs`, even if invoked after them.
   makeOverridablePythonPackage =
     f:
     lib.mirrorFunctionArgs f (
       origArgs:
       let
-        args = lib.fix (
-          lib.extends (_: previousAttrs: {
-            passthru = (previousAttrs.passthru or { }) // {
-              overridePythonAttrs = newArgs: makeOverridablePythonPackage f (overrideWith newArgs);
-            };
-          }) (_: origArgs)
-        );
-        result = f args;
-        overrideWith = newArgs: args // (if pkgs.lib.isFunction newArgs then newArgs args else newArgs);
+        result = f origArgs;
+        overrideWith = newArgs: origArgs // lib.toFunction newArgs origArgs;
+        # Change the result of the function call by applying `transform` to it
+        overrideResult = transform: makeOverridablePythonPackage (args: transform (f args)) origArgs;
```

Comment:
Nit: "override result" is a noun that doesn't really convey that this is a function; it sounds more like a static value. I guess it could mean "override the result", but I interpreted it as "the result of overriding".

We could name it something like "wrapOverride" or "applyTransform"?